ONSdigital/census-rh-ui
# CR-681 RHUI Unlinked UAC journey

# Motivation and Context
Fulfils CR-681 and CR-890, to add the unlinked journey to RHUI

CANNOT BE MERGED WITHOUT MATCHING CUCUMBER CHANGES

# What has changed
New journey triggered by entering a UAC in the 'start' page that is not linked to a Case (specifically the UAC has no value for caseId)
User is prompted for an address, via the 'Enter Address' screen, on to 'Select Address' and on to 'Confirm Address', in the same way as Fulfilment journeys.
If address confirm is 'Yes', new RHSvc linking endpoint is triggered, with payload values from the AIMS UPRN return for the selected address.
If linking successful, user gets a page saying it was successful, with a button to trigger to EQ survey launched (or to NI language question if region is N)
Updates have been made to cover the adlocation param as well

Changes have been made to make the enter/select/confirm address handlers common, so that they can be used in both fulfilment and start based journeys, with some minor changes to the outputs in the fulfilment journeys.

Various elements have been renamed to match region value rather than language (this was incorrect before)

AIMS and RHSvc calls have been moved in to utils as common functions

Note that implementation does not fully match design work, as this has the existing address selection (typeahead address selection is not yet signed off), and includes messaging informing the user why they are being asked for their address.

# How to test?
Unit tests have been updated to cover new functionality, and make various test data correctly in line with reality
There is a matching update to Cucumber to cover page display tweaks in Confirm Address
Missing and fuller testing for adlocation has been added (adlocation testing previously did not check that the value reached EQ payload)
Some start unit tests have been improved and merged to give better and simpler coverage

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-681
https://collaborate2.ons.gov.uk/jira/browse/CR-890

# Screenshots (if appropriate):
